pandas/_libs/tslibs/nattype.pyx

```diff
@@ -540,7 +542,25 @@ NaT = NaTType()
 
 # ----------------------------------------------------------------------
 
-cdef inline bint _checknull_with_nat(object val):
```

Comment:
why don't you put these 2 in a new _libs.dtypes.inference.pyx (and can move things to it later)

Reply:
Would it be OK to keep `checknull_with_nat` with `nattype` and just move `check_all_nulls`?  One of the _really_ nice things about `tslibs` right now is that -- `util` notwithstanding, it is self-contained.

Reply:
but that's exactly the point. this function simply does not belong here. you need another module to control inference. I would implmenet both of these there. tslib will depend on it.

Reply:
Ugh, OK.